container/factory.go

```diff
@@ -27,7 +27,7 @@ import (
 
 type ContainerHandlerFactory interface {
 	// Create a new ContainerHandler using this factory. CanHandleAndAccept() must have returned true.
-	NewContainerHandler(name string, inHostNamespace bool) (c ContainerHandler, err error)
+	NewContainerHandler(name string, metadataEnvs []string, inHostNamespace bool) (c ContainerHandler, err error)
```

Comment:
can we rename `metadataEnvs` to `metadataEnvAllowList` or something like that? From the name, it's not clear that we are only going envs in `metadataEnvs`

Reply:
Sure.